client/me/purchases/utils.ts

```diff
@@ -22,63 +26,46 @@ function canEditPaymentDetails( purchase ) {
 	);
 }
 
-function getChangePaymentMethodPath( siteSlug, purchase ) {
-	if ( isPaidWithCreditCard( purchase ) ) {
-		const {
-			payment: { creditCard },
-		} = purchase;
-
-		return changePaymentMethod( siteSlug, purchase.id, creditCard.id );
+export function getChangePaymentMethodPath( siteSlug: string, purchase: Purchase ) {
+	if ( isPaidWithCreditCard( purchase ) && purchase.payment.creditCard ) {
+		return changePaymentMethod( siteSlug, purchase.id, purchase.payment.creditCard.id );
 	}
 
 	return addPaymentMethod( siteSlug, purchase.id );
 }
 
-function getAddNewPaymentMethodPath() {
+export function getAddNewPaymentMethodPath() {
 	return addNewPaymentMethod;
 }
 
-function isTemporarySitePurchase( purchase ) {
+export function isTemporarySitePurchase( purchase: Purchase ) {
```

Comment:
Should this be returning a string? If so, should be typed as such? When is it best to type the return and when not?

Reply:
Hopefully I was able to explain this somewhat above. TLDR when converting existing functions, IMO it's ok to leave off return types. It's way more important to have a file be using TypeScript than for it to have perfect types.